### ISSUE TITLE
Navigating Google docs text area contents using NVDA.

### ISSUE THREAD

**Issue by khsbory** [nvaccess/nvda#6856]
Hello, I'm Hyongsop Kim in Korea.
In Google Docs, if I type some lines of English text and navigating the linnes, there is no problem.
However if I type some lines of Korean text, NVDA doesn't read the text once I move arrow keys.
It is tested all browsers.
If let NVDA read the lines, I have to move to another window by pressing alt + tab and back again.
And if I type some lines Korean again, the same problem is occurred.
Because of this issue, using Google docs with Korean language is very difficult.
Could you check this issue as soon as possible?

Thank you.

**Comment by jcsteh** [comment]
Can you please try the following:

1. Type some Korean text in Google Docs.
2. Press enter.
3. Try to read by line. I assume this won't work.
4. Press NVDA+f1. This will open the NVDA Log Viewer.
5. Press shift+control+end to select to the end of the text and control+c to copy. Then, paste this text into your reply.

Do you have Google Docs braille mode enabled? If not, please give this a try. Although it's primarily for braille display users, it completely changes the way that Google Docs accessibility works behind the scenes, so it might help. See [this Google article](https://support.google.com/docs/answer/6057417?hl=en) for information about how to enable this.

Thanks.

**Comment by khsbory** [comment]
Thank you for your kind reply.
As you said, I already tried to select braille on mode, but the problem is same.
And as you said, below is the log that I have.
If you need more information, let me know.
 
INFO - __main__ (12:19:28):
Starting NVDA
INFO - core.main (12:19:28):
Config dir: C:\Users\Giyeon Hwang\AppData\Roaming\nvda
INFO - core.main (12:19:28):
NVDA version 2016.4
INFO - core.main (12:19:28):
Using Windows version 10.0.14393 workstation
INFO - core.main (12:19:28):
Using Python version 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:19:22) [MSC v.1500 32 bit (Intel)]
INFO - core.main (12:19:28):
Using comtypes version 0.6.2
INFO - external:synthDrivers.vocalizer.SynthDriver.__init__ (12:19:29):
Vocalizer info: Nuance Vocalizer 5.5, driver version 2.0.2, licensed:C:\Users\Giyeon Hwang\AppData\Roaming\nvda\vocalizer_license.ini
INFO - synthDriverHandler.setSynth (12:19:29):
Loaded synthDriver vocalizer
INFO - core.main (12:19:29):
Using wx version 3.0.2.0 msw (classic)
INFO - braille.initialize (12:19:29):
Using liblouis version 3.0.0
INFO - braille.BrailleHandler.setDisplayByName (12:19:29):
Loaded braille display driver noBraille, current display has 0 cells.
INFO - brailleInput.initialize (12:19:29):
Braille input initialized
WARNING - core.main (12:19:29):
Java Access Bridge not available
INFO - _UIAHandler.UIAHandler.MTAThreadFunc (12:19:29):
UIAutomation: IUIAutomation3
INFO - core.main (12:19:29):
NVDA initialized
INFO - updateCheck.AutoUpdateChecker._started (12:19:29):
Performing automatic update check
ERROR - scriptHandler.executeScript (12:19:57):
error executing script: &lt;bound method GlobalCommands.script_toggleVirtualBufferPassThrough of &lt;globalCommands.GlobalCommands object at 0x0A860A50&gt;&gt; with gesture u'NVDA+\uc2a4\ud398\uc774\uc2a4'
Traceback (most recent call last):
  File "scriptHandler.pyo", line 187, in executeScript
  File "globalCommands.pyo", line 1234, in script_toggleVirtualBufferPassThrough
  File "treeInterceptorHandler.pyo", line 118, in _set_passThrough
  File "review.pyo", line 117, in setCurrentMode
  File "review.pyo", line 25, in getObjectPosition
AttributeError: 'NoneType' object has no attribute 'makeTextInfo'
ERROR - scriptHandler.executeScript (12:20:00):
error executing script: &lt;bound method Gecko_ia2.script_disablePassThrough of &lt;virtualBuffers.gecko_ia2.Gecko_ia2 object at 0x0A6F9690&gt;&gt; with gesture u'escape'
Traceback (most recent call last):
  File "scriptHandler.pyo", line 187, in executeScript
  File "browseMode.pyo", line 395, in script_disablePassThrough
  File "treeInterceptorHandler.pyo", line 123, in _set_passThrough
AttributeError: 'NoneType' object has no attribute 'treeInterceptor'
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (12:21:32):
Developer info for navigator object:
name: u'\ubb38\uc11c \ub0b4\uc6a9'
role: ROLE_EDITABLETEXT
states: STATE_FOCUSABLE, STATE_EDITABLE, STATE_FOCUSED, STATE_MULTILINE
isFocusable: True
hasFocus: True
Python object: &lt;NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorMozillaIAccessible object at 0x0A6F9790&gt;
Python class mro: (&lt;class 'NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorMozillaIAccessible'&gt;, &lt;class 'NVDAObjects.behaviors.EditableTextWithAutoSelectDetection'&gt;, &lt;class 'NVDAObjects.behaviors.EditableText'&gt;, &lt;class 'editableText.EditableText'&gt;, &lt;class 'NVDAObjects.IAccessible.ia2Web.Editor'&gt;, &lt;class 'NVDAObjects.IAccessible.mozilla.Mozilla'&gt;, &lt;class 'NVDAObjects.IAccessible.ia2Web.Ia2Web'&gt;, &lt;class 'NVDAObjects.IAccessible.IAccessible'&gt;, &lt;class 'NVDAObjects.window.Window'&gt;, &lt;class 'NVDAObjects.NVDAObject'&gt;, &lt;class 'baseObject.ScriptableObject'&gt;, &lt;class 'baseObject.AutoPropertyObject'&gt;, &lt;type 'object'&gt;)
description: u''
location: (8, -9930, 609, 0)
value: None
appModule: &lt;'firefox' (appName u'firefox', process ID 6176) at address a891ef0&gt;
appModule.productName: u'Firefox'
appModule.productVersion: u'51.0.1'
TextInfo: &lt;class 'NVDAObjects.IAccessible.ia2TextMozilla.MozillaCompoundTextInfo'&gt;
windowHandle: 1638964L
windowClassName: u'MozillaContentWindowClass'
windowControlID: 0
windowStyle: 1342177280
windowThreadID: 8284
windowText: u'NetscapeDispatchWnd'
displayText: u''
IAccessibleObject: &lt;POINTER(IAccessible2) ptr=0xbcebbe4 at a711580&gt;
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=1638964, objectID=-4, childID=-3879
IAccessible accName: u'\ubb38\uc11c \ub0b4\uc6a9'
IAccessible accRole: ROLE_SYSTEM_TEXT
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048580)
IAccessible accDescription: u''
IAccessible accValue: None
IAccessible2 windowHandle: 1638964
IAccessible2 uniqueID: -3879
IAccessible2 role: ROLE_SYSTEM_TEXT
IAccessible2 states: IA2_STATE_MULTI_LINE, IA2_STATE_OPAQUE, IA2_STATE_EDITABLE (1544)
IAccessible2 attributes: u'margin-left:0px;text-align:start;text-indent:313px;formatting:block;margin-right:0px;xml-roles:textbox;tag:div;margin-top:0px;margin-bottom:0px;display:block;explicit-name:true;'
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (12:22:21):
Developer info for navigator object:
name: u'\ubb38\uc11c \ub0b4\uc6a9'
role: ROLE_EDITABLETEXT
states: STATE_FOCUSABLE, STATE_EDITABLE, STATE_FOCUSED, STATE_MULTILINE
isFocusable: True
hasFocus: True
Python object: &lt;NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorMozillaIAccessible object at 0x0A6CA5D0&gt;
Python class mro: (&lt;class 'NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorMozillaIAccessible'&gt;, &lt;class 'NVDAObjects.behaviors.EditableTextWithAutoSelectDetection'&gt;, &lt;class 'NVDAObjects.behaviors.EditableText'&gt;, &lt;class 'editableText.EditableText'&gt;, &lt;class 'NVDAObjects.IAccessible.ia2Web.Editor'&gt;, &lt;class 'NVDAObjects.IAccessible.mozilla.Mozilla'&gt;, &lt;class 'NVDAObjects.IAccessible.ia2Web.Ia2Web'&gt;, &lt;class 'NVDAObjects.IAccessible.IAccessible'&gt;, &lt;class 'NVDAObjects.window.Window'&gt;, &lt;class 'NVDAObjects.NVDAObject'&gt;, &lt;class 'baseObject.ScriptableObject'&gt;, &lt;class 'baseObject.AutoPropertyObject'&gt;, &lt;type 'object'&gt;)
description: u''
location: (8, -9930, 609, 0)
value: None
appModule: &lt;'firefox' (appName u'firefox', process ID 6176) at address a891ef0&gt;
appModule.productName: u'Firefox'
appModule.productVersion: u'51.0.1'
TextInfo: &lt;class 'NVDAObjects.IAccessible.ia2TextMozilla.MozillaCompoundTextInfo'&gt;
windowHandle: 1638964L
windowClassName: u'MozillaContentWindowClass'
windowControlID: 0
windowStyle: 1342177280
windowThreadID: 8284
windowText: u'NetscapeDispatchWnd'
displayText: u''
IAccessibleObject: &lt;POINTER(IAccessible2) ptr=0xbcebbe4 at 4d91620&gt;
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=1638964, objectID=-4, childID=-3879
IAccessible accName: u'\ubb38\uc11c \ub0b4\uc6a9'
IAccessible accRole: ROLE_SYSTEM_TEXT
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048580)
IAccessible accDescription: u''
IAccessible accValue: None
IAccessible2 windowHandle: 1638964
IAccessible2 uniqueID: -3879
IAccessible2 role: ROLE_SYSTEM_TEXT
IAccessible2 states: IA2_STATE_MULTI_LINE, IA2_STATE_OPAQUE, IA2_STATE_EDITABLE (1544)
IAccessible2 attributes: u'margin-left:0px;text-align:start;text-indent:425px;formatting:block;margin-right:0px;xml-roles:textbox;tag:div;margin-top:0px;margin-bottom:0px;display:block;explicit-name:true;'
 
 
김 형 섭	 / Hyongsop Kim
(주)엔비전스 / 웹접근성사업팀 팀장	

서울시 종로구 가회동 1-29 
1-29 Gahoe-dong, Jongno-gu, Seoul, Korea
T. +82 70 4900 0895   F. +82 2 313 3645   M. +82 10 5359 5996

http://dialogueinthedark.co.kr
http://cafe.naver.com/dialogueinthedark

 
-----Original Message-----

**Comment by jcsteh** [comment]
With braille mode enabled, what happens if you try to move character by
character instead of line by line?
Also, when you say NVDA doesn't read the line, does it say blank or nothing
at all?

**Comment by khsbory** [comment]
1. Even if I press left or right arrow key, NVDA doesn't read the character.
2. NVDA says blank, whenever I press arrow keys.
 
Thank you. 
 
김 형 섭	 / Hyongsop Kim
(주)엔비전스 / 웹접근성사업팀 팀장	

서울시 종로구 가회동 1-29 
1-29 Gahoe-dong, Jongno-gu, Seoul, Korea
T. +82 70 4900 0895   F. +82 2 313 3645   M. +82 10 5359 5996

http://dialogueinthedark.co.kr
http://cafe.naver.com/dialogueinthedark

 
-----Original Message-----

**Comment by Adriani90** [comment]
@khsbory is this sill an issue in NVDA 2019.1?

**Comment by Adriani90** [comment]
@josephsl could you please try to reproduce this? There have been many changes since las update on this issue.

**Comment by Adriani90** [comment]
We didn't get any updates from the initial author since many years, I am closing as abandoned. if you are still having this issue, please comment on this and we can reopen.
